app/feed/page.tsx

```diff
@@ -1,94 +1,54 @@
-"use client";
-
-import GitHubEvent from "@/components/gh_events/GitHubEvent";
-import { IGitHubEvent, combineSimilarPushEvents } from "@/lib/gh_events";
-import { useEffect, useState } from "react";
-import { scrollTo } from "@/lib/utils";
 import LoadingText from "@/components/LoadingText";
-import { env } from "@/env.mjs";
+import { IGitHubEvent } from "@/lib/gh_events";
+import GitHubEvent from "@/components/gh_events/GitHubEvent";
+import { Octokit } from "octokit";
+import getGitHubAccessToken from "@/lib/getGitHubAccessToken";
+
+const GITHUB_PAT: string = getGitHubAccessToken() || "";
+const GITHUB_ORG: string = process.env.NEXT_PUBLIC_GITHUB_ORG || "";
+
+export const revalidate = 600;
 
 type Props = {
   searchParams: {
     page?: string;
   };
 };
+const octokit = new Octokit({
+  auth: GITHUB_PAT,
+});
 
-function extractPaginationLinks(text: string) {
-  // Ref: https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28#using-link-headers
-  const regex = /<([^>]+)>; rel="([^"]+)"/g;
-  const pageNums = {
-    first: undefined as string | undefined,
-    prev: undefined as string | undefined,
-    next: undefined as string | undefined,
-    last: undefined as string | undefined,
-  };
-
-  let match;
-  while ((match = regex.exec(text)) !== null) {
-    const page = new URL(match[1]).searchParams.get("page");
-    pageNums[match[2] as keyof typeof pageNums] = page
-      ? `/feed?page=${page}`
-      : undefined;
-  }
-
-  return pageNums;
-}
-
-export default function FeedPage({ searchParams }: Props) {
-  const [lastFetchedPage, setLastFetchedPage] = useState(1);
-  const [events, setEvents] = useState<Record<number, IGitHubEvent[]>>({});
-
-  useEffect(() => {
-    fetch(
-      `https://api.github.com/orgs/${env.NEXT_PUBLIC_GITHUB_ORG}/events?per_page=1000&page=${lastFetchedPage}`,
-    )
-      .then((res) => res.json())
-      .then((data: IGitHubEvent[]) => {
-        setEvents((existing) => ({
-          ...existing,
-          page: combineSimilarPushEvents(
-            data.filter(exludeBotEvents).filter(excludeBlacklistedEvents),
-          ),
-        }));
-
-        if (lastFetchedPage !== 1) {
-          const lastEvents = events[lastFetchedPage - 1];
-          scrollTo(`gh-event-${lastEvents[lastEvents.length - 1].id}`);
-        }
-      });
-  }, [events, lastFetchedPage]);
-
-  const allEvents = ([] as IGitHubEvent[]).concat(...Object.values(events));
-
+export default async function FeedPage({ searchParams }: Props) {
+  const events = await octokit.paginate(
+    "GET /orgs/{org}/events",
+    {
+      org: GITHUB_ORG,
+      per_page: 1000,
+    },
+    (response) => {
+      const data = response.data as IGitHubEvent[];
+      const filteredData = data
+        .filter(exludeBotEvents)
+        .filter(excludeBlacklistedEvents);
+      return filteredData;
```

Comment:
Octokit responses are typesafe. Avoid casting it to another type.

```suggestion
      return response.data
        .filter(exludeBotEvents)
        .filter(excludeBlacklistedEvents);
```

Reply:
const data = response.data as IGitHubEvent[];
If we remove this line it will create a `Type error: No overload matches this call.`

Reply:
Previous code didn't use octokit either.

Reply:
Yes you are right , But in this if we remove this line so it's genrate an error on build time .
Also we are passing this in Githubevent  component as a prop to render and it also verify type using IGithubEvebt[]. So we have to also make changes in type.ts right?

Reply:
You won't need to define the types if you inline those functions right?

Existing types can be kept as is as it maybe used elsewhere.